R/rvar-summaries-within-draws.R

```diff
@@ -0,0 +1,160 @@
+# Summaries over array elements, within draws ---------------------------------------------------------
+
+#' Summaries of random variables over array elements, within draws
+#'
+#' Compute summaries of random variables over array elements and within draws,
+#' producing a new random variable of length 1 (except in the case of
+#' `rvar_range()`, see 'Details').
+#'
+#' @param ... [`rvar`]s
+#' @template args-rvar-summaries-na.rm
+#' @param constant For `rvar_mad()`, a scale factor for computing the median
+#' absolute deviation. See `mad()`.
+#' @param low For `rvar_mad()`, if `TRUE`, compute the 'lo-median', i.e., for
+#' even sample size, do not average the two middle values, but take the smaller
+#' one. See `mad()`.
+#' @param high For `rvar_mad()`, if `TRUE`, compute the 'hi-median', i.e., take
+#' the larger of the two middle values for even sample size. See `mad()`.
+#'
+#' @details
+#'
+#' These functions compute statistics within each draw of the random variable.
+#' For summaries over draws (such as expectations), see [rvar-summaries-over-draws].
+#'
+#' Each function defined here corresponds to the base function of the same name
+#' without the `rvar_` prefix (e.g., `rvar_mean()` calls `mean()` under the hood, etc).
+#'
+#' @return
+#' An [`rvar`] of length 1 (or in the case of `range()`, length 2) with the same number
+#' of draws as the input rvar(s) containing the summary statistic computed within
+#' each draw of the input rvar(s).
+#'
+#' @examples
+#'
+#' set.seed(5678)
+#' x = rvar_rng(rnorm, 4, mean = 1:4, sd = 2)
+#'
+#' # These will give similar results to mean(1:4),
+#' # median(1:4), sum(1:4), prod(1:4), etc
+#' rvar_mean(x)
+#' rvar_median(x)
+#' rvar_sum(x)
+#' rvar_prod(x)
```

Comment:
add `rvar_range` example to illustrate the special case output?

Reply:
ah good point, done